src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,181 @@ class DocumentView(BaseLTIView):
     serializer_class = DocumentSerializer
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTIRespondView(TemplateResponseMixin, View):
+    """LTI view called to respond to a consumer.
+
+    ie. after a deep linking content selection.
+
+    """
+
+    template_name = "core/form_autosubmit.html"
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
+
+        Renders a form autosubmitted to a LTI consumer with signed parameters.
+
+        Parameters
+        ----------
+        request : Request
+            passed by Django
+        args : list
+            positional extra arguments
+        kwargs : dictionary
+            keyword extra arguments
+        Returns
+        -------
+        HTML
+            generated from applying the data to the template
+
+        """
+        content_item_return_url = self.request.POST.get("content_item_return_url")
+
+        # filters out oauth parameters
+        lti_parameters = {
+            key: value
+            for (key, value) in self.request.POST.items()
+            if "oauth" not in key and key not in ("content_item_return_url",)
+        }
+
+        # generate signature
+        lti = LTI(self.request)
```

Comment:
shouldn't you check the validity of this LTI request first? If yes, add the missing test because it is a vulnerability?

Reply:
I choose no tot do so, because initial LTI request has already been verified, and this request doesn't come from LTi consumer, but from our frontend.
So HTTP_REFERER would fail verification.

Reply:
well I think you still need some authentication before you provide LTI signature computation for free :scream:

Reply:
Ok, I will add an argument to lti.verify(), something like
```
def verify(self, bypass_host_verification=False):
```
WDYT ?

Reply:
After a test through moodle, adding a verification with a simple lti.verify() works well.

Reply:
nice.

Reply:
I'm adding a test, though.